src/usage/vector-types.md

```diff
@@ -43,3 +51,112 @@ We upsert 1M binary vectors into the table and then run a KNN query for each emb
 ![bvector](./images/bvector.png)
 
 We can see that the `bvector`'s accuracy is not as good as the `vector` type, but it exceeds 95%  if we adopt [adaptive retrieval](/use-case/adaptive-retrieval).
+
+## `svector` sparse vector
+
+Unlike dense vectors, sparse vectors are very high-dimensional but contain few non-zero values.
+
+Typically, sparse vectors can be created from:
+- Word co-occurrence matrices
+- Term frequency-inverse document frequency (TF-IDF) vectors
+- User-item interaction matrices
+- Network adjacency matrices
+
+Sparse vectors in `pgvecto.rs` are called `svector`.
+
+::: tip
+`svector` is 32-bit float, 16-bit float sparse vector is not supported yet.
+:::
+
+Here's an example of creating a table with a svector column and inserting values:
+
+```sql {3}
+CREATE TABLE items (
+  id bigserial PRIMARY KEY,
+  embedding svector(10) NOT NULL
+);
+
+INSERT INTO items (embedding) VALUES ('[0.1,0,0,0,0,0,0,0,0,0]'), ('[0,0,0,0,0,0,0,0,0,0.5]');
+```
+
+We support three operators to calculate the distance between two `svector` values.
+
+- `<->` (`svector_l2_ops`): squared Euclidean distance, defined as $\Sigma (x_i - y_i) ^ 2$.
+- `<#>` (`svector_dot_ops`): negative dot product, defined as $- \Sigma x_iy_i$.
+- `<=>` (`svector_cos_ops`): cosine distance, defined as $1 - \frac{\Sigma x_iy_i}{\sqrt{\Sigma x_i^2 \Sigma y_i^2}}$.
+
+Index can be created on `svector` type as well.
+
+```sql
+CREATE INDEX your_index_name ON items USING vectors (embedding svector_l2_ops);
+
+SELECT * FROM items ORDER BY embedding <-> '[0.3,0,0,0,0,0,0,0,0,0]' LIMIT 1;
+```
+
+### Data constructor
+
+There is an index and value constructor to build arbitrary `svector`:
+```sql
+SELECT to_svector(5, '{0,4}', '{0.3,0.5}');
+-- [0.3, 0, 0, 0, 0.5]
+```
+
+## `vecf16` half-precision vector
+
+Stored as a half-precision number format, `vecf16` takes advantage of the 16-bit float, which uses half the memory and bandwidth compared to `vector`.
+It is often faster than the regular `vector` datatype, but may lose some precision.
+
+Here's an example of creating a table with a vecf16 column and inserting values:
+
+```sql {3}
+CREATE TABLE items (
+  id bigserial PRIMARY KEY,
+  embedding vecf16(3) NOT NULL
+);
+
+INSERT INTO items (embedding) VALUES ('[0.1, 0.2, 0]'), ('[0, 0.1, 0.2]');
+```
+
+We support three operators to calculate the distance between two `vecf16` values.
+
+- `<->` (`vecf16_l2_ops`): squared Euclidean distance, defined as $\Sigma (x_i - y_i) ^ 2$.
+- `<#>` (`vecf16_dot_ops`): negative dot product, defined as $- \Sigma x_iy_i$.
+- `<=>` (`vecf16_cos_ops`): cosine distance, defined as $1 - \frac{\Sigma x_iy_i}{\sqrt{\Sigma x_i^2 \Sigma y_i^2}}$.
+
+Index can be created on `vecf16` type as well.
```

Comment:
It's strange because we already list opclasses. We should put the statements before them.